pkg/cnpg/adapter.go

```diff
@@ -130,12 +163,73 @@ func (adapter *CNPGAdapter) ApplyConfig(proposedConfig *agent.ProposedConfigResp
 		return nil
 	}
 
+	// Get current configuration from PostgreSQL to compare
+	// This prevents unnecessary restarts by only applying parameters that actually changed
+	logger.Info("Fetching current configuration from database for comparison")
+	currentConfig, err := adapter.GetCurrentConfig(ctx)
+	if err != nil {
+		logger.Warnf("Failed to get current config for comparison: %v. Will apply all parameters.", err)
+		currentConfig = make(map[string]string) // Empty map = apply all
+	} else {
+		logger.Infof("Successfully retrieved %d current parameter values", len(currentConfig))
+	}
```

Comment:
better bail out here. If there's something wrong, the best thing to do is nothing.